src/components/Executions/ExecutionDetails/NodeExecutionDetailsPanelContent.tsx

```diff
@@ -238,6 +308,26 @@ export const NodeExecutionDetailsPanelContent: React.FC<NodeExecutionDetailsProp
 
     const nodeExecution = nodeExecutionQuery.data;
 
+    const getWorkflowDag = async () => {
+        const workflowExecution = await fetchWorkflowExecution(
+            queryClient,
+            nodeExecutionId.executionId
+        );
+        const workflowData: Workflow = await fetchWorkflow(
+            queryClient,
+            workflowExecution.closure.workflowId
+        );
+        if (workflowData) {
+            const keyedDag = transformWorkflowToKeyedDag(workflowData);
+            setDag(keyedDag);
+        }
+    };
```

Comment:
This is async function, which means it could execute `setDag` after component was already unmounted -> which would lead to a memory leak on the site (see article [here](https://dev.to/alexandrudanpop/correctly-handling-async-await-in-react-components-4h74))

This is pattern I have successfully used so far:
```
useEffect(()=>{
  let isCurrent = true;
  getWorkflowDag().then((keyedDag)=>{
    if (isCurrent) {
      setDag(keyedDag);
    }
  })
  // This handles the unmount case
  return ()=>{ isCurrent = false; }; 
}, [nodeExecutionId])
```

Reply:
Pushed the fix.